frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -3455,6 +3458,190 @@ def reinterpret(self, x: Conversion, *, bit_width: int) -> Conversion:
         )
         return self.operation(operation, resulting_type, x.result)
 
+    def where(
+        self,
+        resulting_type: ConversionType,
+        condition: Conversion,
+        when_true: Conversion,
+        when_false: Conversion,
+    ) -> Conversion:
+        if condition.original_bit_width != 1 or condition.is_signed:
```

Comment:
does signed means it can effectively be negative or it could be added to a signed value only ? if signed but in domain [0, 1] this could be accepted ?

Reply:
If the domain is [0, 1] it will exactly be uint1 in this check. Signed means it will use 2s complement representation, so yes it can be negative for all bit widths.